src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,73 @@
-import React from 'react';
+import React, { useLayoutEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useDispatch } from 'react-redux';
+import { currentTodoSlice } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo;
+}
+
+export const TodoModal: React.FC<Props> = ({ todo }) => {
+  const [isLoading, setLoading] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+  const { removeCurrentTodo } = currentTodoSlice.actions;
+
+  const dispatch = useDispatch();
+
+  useLayoutEffect(() => {
+    getUser(todo.userId)
+      .then(setUser)
+      .catch()
+      .finally(() => setLoading(false));
+  }, [isLoading, todo.userId]);
```

Comment:
Using `isLoading` in the dependency array of `useLayoutEffect` is incorrect because it will cause the effect to run on every render after `isLoading` changes. Consider using an empty dependency array `[]` to run the effect only once when the component mounts.